astropy_healpix/tests/test_core.py

```diff
@@ -28,6 +30,35 @@ def test_level_to_nside():
     assert exc.value.args[0] == 'level must be positive'
 
 
+def test_nside_to_level():
+    assert nside_to_level(1024) == 10
+    with pytest.raises(ValueError) as exc:
+        nside_to_level(511)
+    assert exc.value.args[0] == 'nside must be a power of two'
+
+
+def test_levelipix_to_uniq():
+    assert 11 + 4*4**0 == levelipix_to_uniq(0, 11)
+    assert 62540 + 4*4**15 == levelipix_to_uniq(15, 62540)
+    with pytest.raises(ValueError) as exc:
+        levelipix_to_uniq(1, 49)
+    assert exc.value.args[0] == 'ipix for a specific level must be inferior to npix'
+
+
+@pytest.mark.parametrize("level", [
+    0, 5, 10, 15, 20, 22, 25, 26, 27, 28, 29
+])
+def test_uniq_to_levelipix(level):
+    # Generate 10 HEALPix indexes for each level
+    size = 10
+    npix = 3 << 2*(level + 1)
+    ipix_arr = np.random.randint(npix, size=size, dtype=np.int64)
```

Comment:
Suggest to remove the `_arr` from the variable names here. They make the lines very long and don't help make the code more readable IMO.

Reply:
@bmatthieu3 - Can you get rid of random numbers here completely? If yes, please do.
If no, please use a seed and avoid changing the global `np.random` stream.
I think we already have examples how to do that in other parts of the tests of astropy-healpix, but if not and it's not clear what I mean, let me know and I'll point out an example.

Reply:
@cdeil Ok I can remove this call to ``np.random``. Just to know, what do you mean by changing the global np.random stream ? and especially why do we do not want this ?
Is this for the following reason ? :
Some tests that are called after mine are waiting for specific rand results (based on the default internal seed maybe) that would change if I make a new call to ``np.random.randint`` before them ?

Reply:
If you just call functions like `np.random.random()` from all over the place (i.e. separate tests), then you will get different numbers depending on test order execution or just the machine you're on.
Then you might get numbers and failing tests at arbitrary times, which is annoying.

You can call `np.random.seed(0)` each time before generating random numbers, or create a `rng = RandomState(seed)` and call the methods on that, or use `with NumpyRNGContext(seed)`. It seems in Astropy all of those patterns are used:
https://github.com/astropy/astropy/search?q=random&unscoped_q=random

Looks like in astropy-healpix we already have one "rogue" `np.random` call, that should be cleaned up as well:
https://github.com/astropy/astropy-healpix/blob/528292bc2bff01563d6c696880020345cde88d40/astropy_healpix/tests/test_healpy.py#L285

IMO the best solution is to not use random numbers at all in almost all cases. If you write an explicit list of numbers as test cases and manually choose 1 - 10 numbers well to cover what you want to test, it's just better than producing random numbers, becuase it's more explicit and you know what is being tested, e.g. which range of numbers.